File: 3- Remove duplicates/Java/solution3.java
// By using a dictionary:

// Time complexity: O(n)
// Space complexity: O(n)

static int[] removeDuplicates(int[] arr){
  HashMap<Integer,Boolean> visited = new HashMap<Integer,Boolean>();
  for(int element : arr) 
    visited.put(element, true);
  int[] noDuplicatesArr = new int[visited.keySet().size()];
  int i = 0;
  for(Integer key : visited.keySet())
    noDuplicatesArr[i++] = key;
  return noDuplicatesArr;
}

File: 3- Remove duplicates/C++/solution3.cpp
// By using a dictionary:

// Time complexity: O(n)
// Space complexity: O(n)

vector<int> removeDuplicates(vector<int> arr){
  unordered_map<int,bool> visited;
  for(int i = 0; i < arr.size(); i++) 
    visited[arr[i]] = true;
  vector<int> noDuplicatesArr;
  for(auto const& pair : visited)
    noDuplicatesArr.push_back(pair.first);
  return noDuplicatesArr;
}


